### PR TITLE
Dim/type change cach invalidation

### PR DESCRIPTION
**Pull request by didimitrie**
[CNX-805: Changing an ElementType should flag all objects of that type as changed](https://linear.app/speckle/issue/CNX-805/changing-an-elementtype-should-flag-all-objects-of-that-type-as)

Tested in 2023, 2024 and 2025

https://github.com/user-attachments/assets/906ffcd7-1bd9-481f-ba32-a9b266400864

Tested with the revit performance model as well, feels ok - but we defs need to get the idle manager to call things once asap-ish.

### CODE REVIEW

File: Connectors/Revit/Speckle.Connectors.RevitShared/Bindings/RevitSendBinding.cs [hunk 1]
```diff
@@ -346,15 +346,33 @@ private async Task RunExpirationChecks()
     }
 
     var objUniqueIds = new List<string>();
+    var changedIds = ChangedObjectIds.Keys.ToList();
+
+    // Handling type changes: if an element's type is changed, we need to mark as changed all objects that have that type.
+    // Step 1: get any changed types
+    var elementTypeIdsList = changedIds
+      .Select(e => doc.GetElement(e))
+      .Where(el => el is ElementType)
```
Comment (adamhathcock) [inline Connectors/Revit/Speckle.Connectors.RevitShared/Bindings/RevitSendBinding.cs:355]: OfType? https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.oftype?view=net-9.0

File: Connectors/Revit/Speckle.Connectors.RevitShared/Bindings/RevitSendBinding.cs [hunk 2]
```diff
@@ -346,15 +346,33 @@ private async Task RunExpirationChecks()
     }
 
     var objUniqueIds = new List<string>();
+    var changedIds = ChangedObjectIds.Keys.ToList();
```
Comment (adamhathcock) [inline Connectors/Revit/Speckle.Connectors.RevitShared/Bindings/RevitSendBinding.cs:349]: Perf: Do we need to make another collection here?

File: Connectors/Revit/Speckle.Connectors.RevitShared/Bindings/RevitSendBinding.cs (hunk 1 above)
Comment (didimitrie) [reply Connectors/Revit/Speckle.Connectors.RevitShared/Bindings/RevitSendBinding.cs:360]: legit, changing

File: Connectors/Revit/Speckle.Connectors.RevitShared/Bindings/RevitSendBinding.cs (hunk 2 above)
Comment (didimitrie) [reply Connectors/Revit/Speckle.Connectors.RevitShared/Bindings/RevitSendBinding.cs:354]: i am adding to it below, so i kinda need it in there

**Review by oguzhankoral — APPROVED** [review]

**Pull request merged**